lib/routes/root.js

```diff
@@ -31,16 +31,24 @@ module.exports = {
       return sendReply();
     }
 
-    // figure it out from .git
-    var gitDir = path.resolve(__dirname, '..', '..', '.git');
-    exec('git rev-parse HEAD', { cwd: gitDir }, function(err, stdout) {
-      commitHash = stdout.replace(/\s+/, '');
-      var configPath = path.join(gitDir, 'config');
-      var cmd = 'git config --get remote.origin.url';
-      exec(cmd, { env: { GIT_CONFIG: configPath, PATH: process.env.PATH }}, function(err, stdout) {
-        source = stdout.replace(/\s+/, '');
-        return sendReply();
+    function runGitCmd() {
+      return new Promise((resolve) => {
+        // figure it out from .git
+        var gitDir = path.resolve(__dirname, '..', '..', '.git');
+        exec('git rev-parse HEAD', { cwd: gitDir }, function(err, stdout) {
```

Comment:
Should this reject if an error is found?

Reply:
Might be able to clean up some of callback nesting with http://bluebirdjs.com/docs/api/promise.fromcallback.html, but meh.

Reply:
I think in this case we just don't report the information for the `source` and etc